modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java

```diff
@@ -1,141 +1,353 @@
 package de.wuespace.telestion.api.message;
 
-import com.fasterxml.jackson.annotation.JsonProperty;
 import io.vertx.core.Future;
 import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
 import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.EncodeException;
+import io.vertx.core.json.Json;
 import io.vertx.core.json.JsonObject;
-import io.vertx.core.json.jackson.JacksonCodec;
 import io.vertx.core.spi.json.JsonCodec;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * The base class for all messages which are automatically encoded with the JsonMessageCodec.<br>
+ * <h2>Description</h2>
+ * The base class for all messages which are automatically encoded with the JsonMessageCodec.
+ * <p>
  * All subclasses have to be valid json classes. This means that they could be encoded by
- * {@link io.vertx.core.spi.json.JsonCodec} which is backed by {@link io.vertx.core.json.jackson.JacksonCodec}.
+ * {@link JsonCodec} which is backed by {@link io.vertx.core.json.jackson.JacksonCodec JacksonCodec}.
  *
- * @author Jan von Pichowski (@jvpichowski), Cedric Boes (@cb0s)
- * @version 1.2
+ * <h2>Usage</h2>
+ * <pre>
+ * {@code
+ * public record TimeMessage(
+ *     @JsonProperty long receiveTime,
+ *     @JsonProperty long sendTime
+ * ) implements JsonMessage {
+ * }
+ * }
+ * </pre>
+ *
+ * @author Jan von Pichowski (@jvpichowski), Cedric Boes (@cb0s), Ludwig Richter (@fussel178)
  */
 public interface JsonMessage {
-	JsonCodec JSON_CODEC = new JacksonCodec();
-	Logger logger = LoggerFactory.getLogger(JsonMessage.class);
+
+	///////////////////////////////////////////////////////////////////////////
+	// asynchronous decoding section
+	///////////////////////////////////////////////////////////////////////////
+
+	/**
+	 * Asynchronous version of {@link #from(Buffer, Class)}.
+	 *
+	 * @param type             the class of the target {@link JsonMessage}
+	 * @param json             the buffer whose contents contain the necessary information to construct
+	 *                         the specified {@link JsonMessage}
+	 * @param handler          gets called when the conversion was successful
+	 * @param exceptionHandler gets called when a {@link DecodeException} occurred during conversion
+	 * @param <T>              the type of the target {@link JsonMessage}
+	 * @return {@code true} when the conversion was successful
+	 */
+	static <T extends JsonMessage> boolean on(Class<T> type, Buffer json, Handler<T> handler,
+											  Handler<RuntimeException> exceptionHandler) {
+		try {
+			handler.handle(from(json, type));
+			return true;
+		} catch (DecodeException | IllegalArgumentException e) {
+			logger.warn("Cannot convert buffer to JsonMessage {}:", type.getName(), e);
```

Comment:
@pklaschka what do you think? Are those messages a little short or are they according to AWESA?